### ISSUE TITLE
separate shard and replicaset names

### ISSUE THREAD

**Issue by shortdudey123** [sous-chefs/mongodb#143]
Copied from https://github.com/edelight/chef-mongodb/issues/95

```
If choosing between two recipes that accomplish the same thing, but one is simpler and the other is more complicated... which is better? I am thinking about the idea that the variable shard_name may be extraneous, and that it complicates usage of this recipe by adding yet another variable which you have to think about and understand. When setting up a shard manually (without chef), do you usually assign shard_name? A shard is either a replica set, or it's a single machine. What makes sense is to name the replica set with a replicaset_name. And just that. Only that. Perhaps I am missing something though.
```

tl; dr is shard_name really needed?

**Comment by damacus** [comment]
Closing as superseded by #336. Sharding and replica-set behavior now belongs to the new resource API rather than the removed legacy definition/recipe implementation. Please open a fresh issue if the current resources still need this capability.
